### DISCUSSION TITLE
Current download progress reflects single download instead of total

### DISCUSSION THREAD

**Discussion by Nachtalb** [nicotine-plus/nicotine-plus#2370] [General]
![image](https://user-images.githubusercontent.com/9467802/216802722-50127401-99fc-48f8-b9e0-aa58f3a271f8.png)

The x of the "x MiB / y MiB" download progress at the top is constantly set to the same value as the current downloaded file, thus not reflecting the overall state as the graph would lead one to believe. 

Additionally, the y side decreases after each file that finished downloading, thus again not reflecting the overall progress but the size left to download, which does not match up with the rest of the interface.

Status after writing this issue to see the change in x and y: 
![image](https://user-images.githubusercontent.com/9467802/216802869-bec37fba-114b-421e-b0f0-5850bb6b2c9a.png)

⚠️ Am using the latest [master](https://github.com/nicotine-plus/nicotine-plus/commit/3e54a7e7f48526f1faff070f381d34ccfde9cbda).  But I still made the issue to catch it early on :) 

PS: Props on the new design. It looks freaking great 👍 !!

**Comment by mathiascode** [comment]
Are those filtered statuses incorrect? We don't count filtered files in the total progress, and it shouldn't be possible for files to reach 100% and have the filtered status.

> PS: Props on the new design. It looks freaking great +1 !!

Thanks!

**Reply by Nachtalb** [reply]
The filtered files are ignored correctly; the problem is not with them. Let me show you in a small screen capture: 

....

As I wanted to create the screen capture, the problem does not occur anymore..... I entirely quit nicotine yesterday and started it up from new. 

I want to test the behaviour again with the same environment as yesterday (downloading the same files as then), but the user I downloaded the files from is currently not online. So I guess we can put this ticket on hold for now. 

--- 

To clarify the problem, I had a bit better, maybe. 
As seen in the screenshot, it's a tree structure, whereas the top entry is the summary of all items beneath it. Thus, it shows the total `downloaded in total (mutable) / total to download (immutable)`, which it usually does. But in the case yesterday, it showed not the _total it downloaded_ but _how much it downloaded of the currently transferring file_ and the _total to download_ did not reflect the whole size of all downloaded and to be downloaded files but _only the to be downloaded files_. Thus it wasn't `downloaded in total (mutable) / total to download (immutable)` but `downloaded of the current transferring file (mutable) / total to be downloaded - already downloaded (mutable)`. 

It's a bit here and there, but I hope you understand what I mean XD.

**Reply by mathiascode** [reply]
I understand what you meant, but I don't know why it would behave like this when downloads aren't autocleared. The only special case in the code is the "Filtered" status, so I'm curious if it resulted in other files being skipped somehow: https://github.com/nicotine-plus/nicotine-plus/blob/980ee325a035a5d40471b3ebeb9fd368c96aca6e/pynicotine/gtkgui/transferlist.py#L403-L406

Actually, looking at your screenshot again, I just realized that this code is most likely skipping the folder rows as well, while it should only skip file rows.

**Reply by mathiascode** [reply]
Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/23180d9623e9c98f5cb2ad2d4adf877d07e6d7f0
